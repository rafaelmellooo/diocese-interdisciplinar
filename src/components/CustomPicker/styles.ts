import { StyleSheet } from 'react-native';

export const styles = StyleSheet.create({
    container: {
        flexDirection: 'row',
        alignItems: 'center',
        paddingHorizontal: 10
    },
    pickerContainer: {
        flex: 1
    }
});