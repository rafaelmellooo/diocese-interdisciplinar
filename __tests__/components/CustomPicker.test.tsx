import React from "react";
import { render, fireEvent } from "@testing-library/react-native";
import renderer from "react-test-renderer";
import CustomPicker from "../../src/components/CustomPicker";

describe("CustomPicker component", () => {
  it("renders correctly", () => {
    const mockOnCityChange = jest.fn();
    const tree = renderer
      .create(<CustomPicker onChange={mockOnCityChange} />)
      .toJSON();

    expect(tree).toMatchSnapshot();
  });

//   it("calls onCityChange when a city is selected", () => {
//     const mockOnCityChange = jest.fn();
//     const { getByLabelText } = render(
//       <CustomPicker onChange={mockOnCityChange} />
//     );

//     fireEvent.changeText(
//       getByLabelText("Selecione uma cidade"),
//       "Bertioga - SP"
//     );

//     expect(mockOnCityChange).toHaveBeenCalledWith(60);
//   });
});
