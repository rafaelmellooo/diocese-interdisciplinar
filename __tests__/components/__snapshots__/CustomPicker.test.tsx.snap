// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CustomPicker component renders correctly 1`] = `
<View
  style={
    [
      {
        "alignItems": "center",
        "flexDirection": "row",
        "paddingHorizontal": 10,
      },
      {
        "backgroundColor": "rgb(242, 242, 242)",
      },
    ]
  }
>
  <Text />
  <View
    style={
      [
        {
          "flex": 1,
        },
        {
          "color": "rgb(28, 28, 30)",
        },
      ]
    }
  >
    <RNCPicker
      items={
        [
          {
            "label": "Selecione uma cidade",
            "testID": undefined,
            "textColor": undefined,
            "value": null,
          },
          {
            "label": "Bertioga - SP",
            "testID": undefined,
            "textColor": undefined,
            "value": 60,
          },
          {
            "label": "Cubatão - SP",
            "testID": undefined,
            "textColor": undefined,
            "value": 57,
          },
          {
            "label": "Guarujá - SP",
            "testID": undefined,
            "textColor": undefined,
            "value": 58,
          },
          {
            "label": "Itanhaém - SP",
            "testID": undefined,
            "textColor": undefined,
            "value": 61,
          },
          {
            "label": "Mongaguá - SP",
            "testID": undefined,
            "textColor": undefined,
            "value": 63,
          },
          {
            "label": "Peruíbe - SP",
            "testID": undefined,
            "textColor": undefined,
            "value": 62,
          },
          {
            "label": "Praia Grande - SP",
            "testID": undefined,
            "textColor": undefined,
            "value": 64,
          },
          {
            "label": "Santos - SP",
            "testID": undefined,
            "textColor": undefined,
            "value": 59,
          },
          {
            "label": "São Vicente - SP",
            "testID": undefined,
            "textColor": undefined,
            "value": 56,
          },
        ]
      }
      numberOfLines={1}
      onChange={[Function]}
      selectedIndex={0}
      style={
        [
          {
            "height": 216,
          },
          undefined,
        ]
      }
    />
  </View>
</View>
`;
